plugins/polio/api/vaccines/supply_chain.py

```diff
@@ -185,8 +199,23 @@ def validate(self, attrs: Any) -> Any:
         validated_data = super().validate(attrs)
         if "PO" in validated_data.get("po_number", "") or "po" in validated_data.get("po_number", ""):
             raise serializers.ValidationError("PO number should not be prefixed")
+
+        # Check if user has edit permission, if not raise 403
+        if not self.get_can_edit(VaccineArrivalReport.objects.get(id=attrs["id"])):
```

Comment:
Just as above, I think you should be able to avoid recomputing `get_can_edit()` here.